src/WinGetServer/WinMain.cpp

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include <winrt/Microsoft.Management.Deployment.h>
+#include <wrl/module.h>
+#include <wil/resource.h>
+
+using namespace winrt::Microsoft::Management::Deployment;
+
+CoCreatableClassWrlCreatorMapInclude(PackageManager1);
+CoCreatableClassWrlCreatorMapInclude(PackageManager2);
+CoCreatableClassWrlCreatorMapInclude(FindPackagesOptions1);
+CoCreatableClassWrlCreatorMapInclude(FindPackagesOptions2);
+CoCreatableClassWrlCreatorMapInclude(CreateCompositePackageCatalogOptions1);
+CoCreatableClassWrlCreatorMapInclude(CreateCompositePackageCatalogOptions2);
+CoCreatableClassWrlCreatorMapInclude(InstallOptions1);
+CoCreatableClassWrlCreatorMapInclude(InstallOptions2);
+CoCreatableClassWrlCreatorMapInclude(PackageMatchFilter1);
+CoCreatableClassWrlCreatorMapInclude(PackageMatchFilter2);
+
+// Holds the wwinmain open until COM tells us there are no more server connections
+wil::unique_event _comServerExitEvent;
+
+// Routine Description:
+// - Called back when COM says there is nothing left for our server to do and we can tear down.
+static void _releaseNotifier() noexcept
+{
+    _comServerExitEvent.SetEvent();
+}
+
+int __stdcall wWinMain(_In_ HINSTANCE, _In_opt_ HINSTANCE, _In_ LPWSTR, _In_ int)

```

Comment:
>wWinMain [](http://example.com/codeflow?start=14&length=8)

Needs experimental feature and to respect the overall "can use WPM" policy.

Reply:
The overall policy is here:
[https://github.com/microsoft/winget-cli/blob/862a2af3cd836f682a95093919e4f189f01e4c94/src/AppInstallerCommonCore/Public/winget/GroupPolicy.h#L34](https://github.com/microsoft/winget-cli/blob/862a2af3cd836f682a95093919e4f189f01e4c94/src/AppInstallerCommonCore/Public/winget/GroupPolicy.h#L34)

Reply:
Done.